scripts/src/javatests/org/oppia/android/scripts/todo/TodoCheckTest.kt

```diff
@@ -0,0 +1,422 @@
+package org.oppia.android.scripts.todo
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.oppia.android.scripts.proto.TodoExemption
+import org.oppia.android.scripts.proto.TodoExemptions
+import org.oppia.android.testing.assertThrows
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.PrintStream
+
+/** Tests for [TodoCheck]. */
+class TodoCheckTest {
+  private val outContent: ByteArrayOutputStream = ByteArrayOutputStream()
+  private val originalOut: PrintStream = System.out
+  private val TODO_CHECK_PASSED_OUTPUT_INDICATOR: String = "TODO CHECK PASSED"
+  private val TODO_SYNTAX_CHECK_FAILED_OUTPUT_INDICATOR: String = "TODO CHECK FAILED"
+  private val pathToProtoBinary = "scripts/assets/todo_exemptions.pb"
+
+  @Rule
+  @JvmField
+  var tempFolder = TemporaryFolder()
+
+  @Before
+  fun setUp() {
+    tempFolder.newFolder("testfiles")
+    tempFolder.newFolder("scripts", "assets")
+    tempFolder.newFile(pathToProtoBinary)
+    System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun restoreStreams() {
+    System.setOut(originalOut)
+  }
+
+  @Test
+  fun testTodoCheck_noJsonFilePresent_checkShouldFail() {
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(
+      "${retrieveTestFilesDirectoryPath()}/open_issues.json (No such file or directory)"
```

Comment:
Prefer checking & explicitly throwing in the script vs. relying on the system to do this (otherwise different implementations may vary this error message & cause test flakes).

Reply:
Done.